File: tidyBom.R
# tidyBom.R is a function that cleans data from the Stops-MillStyleYarnBOMs.csv file.

tidyBom <- function(){
        # Load necessary packages
        library(dplyr)
        
        # Read in data and set to local variable 
        bom_tbl <- tbl_df(read.csv("~/Washougal-Mill-Study/CSVs/Stops-MillStyleYarnBOMs.csv"))
        
        # Cleans names of fields
        names(bom_tbl) <- tolower(names(bom_tbl))
        names(bom_tbl) <- sub("\\.", "", names(bom_tbl))
        colnames(bom_tbl)[1] <- "millstyle"
        
        # Removes whitespace and sets millstyle to factor variable
        bom_tbl$millstyle <- gsub(" ", "", bom_tbl$millstyle, fixed = TRUE)
        bom_tbl$millstyle <- as.factor(bom_tbl$millstyle)
        
        # Selects and orders fields from table
        bom_tbl <- 
                bom_tbl %>%
                select(millstyle, yarn, lbsperpiece) 
        
        # Returns table
        return(bom_tbl)
}

File: Python Scripts/tidyBom.py
# tidyBom.py is a function that cleans data from the Stops-MillStyleYarnBOMs.csv file.

def tidyBom():
	# Load libraries
	import pandas as pd

	# Read in data and set to local variable
	bom_filename = "~/Desktop/Pendleton/CSVs/Stops-MillStyleYarnBOMs.csv"
	bom_df = pd.read_csv(bom_filename)

	# Clean names of fields
	bom_df.columns = map(str.lower, bom_df.columns)

	# Renames millstyle field
	bom_df = bom_df.rename(columns = {'millstylekey' : 'millstyle', 'yarn#' : 'yarn'})

	# Remove Whitespace
	bom_df['millstyle'] = bom_df['millstyle'].str.replace(" ", "")

	# Drop Date column
	bom_df = bom_df.drop('datelastused', axis = 1)

	return bom_df